griptape/drivers/file_manager/griptape_cloud_file_manager_driver.py

```diff
@@ -0,0 +1,160 @@
+        bucket_name: The name of the new Bucket to be created if `bucket_id` is not set.
+        workdir: The working directory. List, load, and save operations will be performed relative to this directory.
+        base_url: The base URL of the Griptape Cloud API. Defaults to the value of the environment variable
+            `GT_CLOUD_BASE_URL` or `https://cloud.griptape.ai`.
+        api_key: The API key to use for authenticating with the Griptape Cloud API. If not provided, the driver will
+            attempt to retrieve the API key from the environment variable `GT_CLOUD_API_KEY`.
+
+    Raises:
+        ValueError: If `api_key` is not provided, if `workdir` does not start with "/"", or invalid `bucket_id` and/or `bucket_name` value(s) are provided.
+    """
+
+    bucket_id: Optional[str] = field(default=Factory(lambda: os.getenv("GT_CLOUD_BUCKET_ID")), kw_only=True)
+    bucket_name: Optional[str] = field(default=None, kw_only=True)
+    workdir: str = field(default="/", kw_only=True)
+    base_url: str = field(
+        default=Factory(lambda: os.getenv("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")),
+    )
+    api_key: Optional[str] = field(default=Factory(lambda: os.getenv("GT_CLOUD_API_KEY")))
+    headers: dict = field(
+        default=Factory(lambda self: {"Authorization": f"Bearer {self.api_key}"}, takes_self=True),
+        init=False,
+    )
+
+    @workdir.validator  # pyright: ignore[reportAttributeAccessIssue]
+    def validate_workdir(self, _: Attribute, workdir: str) -> None:
+        if not workdir.startswith("/"):
+            raise ValueError(f"{self.__class__.__name__} requires 'workdir' to be an absolute path, starting with `/`")
+
+    @api_key.validator  # pyright: ignore[reportAttributeAccessIssue]
+    def validate_api_key(self, _: Attribute, value: Optional[str]) -> str:
+        if value is None:
+            raise ValueError(f"{self.__class__.__name__} requires an API key")
+        return value
+
+    def __attrs_post_init__(self) -> None:
+        if self.bucket_id and self.bucket_name:
+            raise ValueError("Only one of 'bucket_id' or 'bucket_name' may be provided, not both.")
+
+        if self.bucket_id:
+            try:
+                self._call_api(method="get", path=f"/buckets/{self.bucket_id}").json()
+            except requests.exceptions.HTTPError as e:
+                if e.response.status_code == 404:
+                    raise ValueError(f"No Bucket found with ID: {self.bucket_id}") from e
+                raise ValueError(f"Unexpected error when retrieving Bucket with ID: {self.bucket_id}") from e
+        elif self.bucket_name:
+            data = {"name": uuid.uuid4().hex} if self.bucket_name is None else {"name": self.bucket_name}
+            post_bucket_response = self._call_api(method="post", path="/buckets", json=data).json()
+            self.bucket_id = post_bucket_response["bucket_id"]
+            logger.info("Created new Bucket with ID: %s", self.bucket_id)
```

Comment:
I don't love the Driver auto-creating a resource like this. What use-case would this enable?

Reply:
Use case is "Upload a file to a new path in the bucket" For S3/local filesystem, there is no need to create an underlying asset but there is for GTC

Reply:
We've made the conscious decision to not auto-create control-plane resources in things like Vector Store Drivers so this does feel like it's going against that decision. Maybe we can include some provisioning sample code in the docs example?

Reply:
I am OK with that for Buckets. But creating an Asset on try_save_file seems valuable to me.

Reply:
Yes definitely on assets. Data plane resources are fine.